### DISCUSSION TITLE
Add overloads for |> to value types that wrap functions

### DISCUSSION THREAD

**Discussion by JaapWijnen** [pointfreeco/swift-prelude#165] [Ideas]
Would it make sense to add overloads for `|>` to types like `Func` and `Endo` (Maybe others, I've mainly used these).
This would clean up the call site since these types represent functions anyway. Then we don't have to call `.call` or `.apply` anymore

given: 
`let increment = Endo<Int> { value in value + 1 }`
before:
`let result = 5 |> increment.call`
with overloads:
`let result = 5 |> increment`

This is not too bad ofcourse but it's really nice when chaining methods for example:
`let result = 5 |> increment <> square <> increment` 
Would have to be wrapped in parentheses:
`let result = 5 |> (increment <> square <> increment).call`

Maybe this also applies to a few other types? I'd gladly add them if you think this would make sense :)

```
public func |> <A> (a: A, f: Endo<A>) -> A {
    return a |> f.call
}

public func |> <A, B> (a: A, f: Func<A, B>) -> B {
    return a |> f.call
}
```

Maybe it also applies to `Parallel`?

**Comment by stephencelis** [comment]
Thanks for bringing this up! This is relevant to #155, I believe.

I'm going to convert to a discussion, since we probably won't address directly in this repo anytime soon, but welcome discussion for downstream users/forks.
